backend/internal/db/dbgorm/dbgorm.go

```diff
@@ -120,3 +122,17 @@ func get[T, R any](db *gorm.DB, ctx context.Context, f func(t T) (*R, error), wh
 
 	return f(r)
 }
+
+func delete[T any](db *gorm.DB, ctx context.Context, t T, where string, args ...any) error {
```

Comment:
I'm not experienced enough with this code to know how this will react if you're trying to delete something which other tables reference. I assume that's why a generic `delete` didn't exist until now. Please check with @efiShtain.

Reply:
gorm by default will use soft delete, e.g. mark objects as deleted and won't fetch them as references.
@efiShtain , please correct me if I'm wrong and if you thin we should do anything else

Reply:
@ig-ra gorm will soft delete only if our model has gorm.DeletedAt type field set, which none of our models has
so if you need soft delete, please add the field (and verify the result)

second thing is about deleting stuff that other reference,
as long as we keep the correct foreign keys and cascading definition (which I'm not sure we have)
I guess this should work

this is a very dangerous function, and should probably 

we should probably test it very well in all cases if we want to keep it here

anyhow,
currently only session delete uses it, so not sure we need the generic part

and you need to check that if you delete a session, it should delete all its data (session call, attempts, what so ever or mark it as soft delete as well)

Reply:
implemented with softdelete. See tests